widget_driver/test/widget_driver_test.dart

```diff
@@ -0,0 +1,118 @@
+
+        final driverTester = await tester.getDriverTester(
+          driverBuilder: (context) => ConcreteWidgetDriver(context),
+        );
+        final driver = driverTester.driver;
+        driver.addListener(() {
+          numberOfDriverListenersEmits += 1;
+        });
+
+        await driverTester.verifyNoMoreCallsToNotifyWidget();
+
+        expect(numberOfDriverListenersEmits, equals(0));
+      });
+
+      testWidgets('Calling notifyWidget once triggers listener to emit once', (WidgetTester tester) async {
+        int numberOfDriverListenersEmits = 0;
+
+        final driverTester = await tester.getDriverTester(
+          driverBuilder: (context) => ConcreteWidgetDriver(context),
+        );
+        final driver = driverTester.driver;
+        driver.addListener(() {
+          numberOfDriverListenersEmits += 1;
+        });
+
+        driver.notifyWidget();
+        await driverTester.waitForNotifyWidget();
+        await driverTester.verifyNoMoreCallsToNotifyWidget();
+
+        expect(numberOfDriverListenersEmits, equals(1));
+      });
+
+      testWidgets('Calling notifyWidget three times triggers listener to emit three times',
+          (WidgetTester tester) async {
+        int numberOfDriverListenersEmits = 0;
+
+        final driverTester = await tester.getDriverTester(
+          driverBuilder: (context) => ConcreteWidgetDriver(context),
+        );
+        final driver = driverTester.driver;
+        driver.addListener(() {
+          numberOfDriverListenersEmits += 1;
+        });
+
+        driver.notifyWidget();
+        driver.notifyWidget();
+        driver.notifyWidget();
+
+        await driverTester.waitForNotifyWidget(numberOfCalls: 3);
+        await driverTester.verifyNoMoreCallsToNotifyWidget();
+
+        expect(numberOfDriverListenersEmits, equals(3));
+      });
+    });
+
+    /*
+      This test cannot run yet since it depends on a new version
+      of `widget_driver_test`. But the `widget_driver_test` needs a new 
+      version of `widget_driver` before it can be deployed :)
+
+      So the goal is:
+      1: Deploy new version of `widget_driver`
+      2: Deploy new version of `widget_driver_test`
+      3: Add this code back.
+
+    group('Lifecycle:', () {
+      testWidgets('Calls dispose when driver gets deallocated', (WidgetTester tester) async {
+        bool disposeWasCalled = false;
+
+        final driverTester = await tester.getDriverTester(
+          driverBuilder: (context) => ConcreteWidgetDriver(context),
+        );
+        final driver = driverTester.driver;
+        driver.disposedCallback = () {
+          disposeWasCalled = true;
+        };
+
+        // Here we put a new widget in the test-screen.
+        // This will remove the widget which contains our
+        /// Change this.. move this into the driverTester.. as a helper function...
+        await tester.pumpWidget(const SizedBox.shrink());
+
+        expect(disposeWasCalled, true);
+      });
+    });
+    */
```

Comment:
Do we really want to commit commented code? Perhaps keep it somewhere else than in github until it can get commited  :-)